# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/frontend/Parser.h
# Commit: d96f98f974e0
# Full Hash: d96f98f974e087224e0a1ef9be67e2fc63397d8b
# Author: Ashley Hauck <khyperia@mozilla.com>
# Date: 2019-03-21 04:34:45
# Regressor Bug: 1535166
# File Overlap Count: 1
# Description:
#   Bug 1535166 - Implement computed field names. r=jorendorff
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D23408
# ==============================================================================

diff -r c786b72a9d8e -r d96f98f974e0 js/src/frontend/Parser.h
--- a/js/src/frontend/Parser.h	Wed Mar 20 15:50:30 2019 +0000
+++ b/js/src/frontend/Parser.h	Wed Mar 20 17:26:01 2019 +0000
@@ -1311,6 +1311,7 @@
                                 HandlePropertyName className,
                                 uint32_t classStartOffset, bool hasHeritage,
                                 size_t& numFieldsWithInitializers,
+                                size_t& numFieldKeys,
                                 ListNodeType& classMembers, bool* done);
   MOZ_MUST_USE bool finishClassConstructor(
       const ParseContext::ClassStatement& classStmt,
@@ -1318,8 +1319,8 @@
       uint32_t classEndOffset, size_t numFieldsWithInitializers,
       ListNodeType& classMembers);
 
-  FunctionNodeType fieldInitializer(YieldHandling yieldHandling,
-                                    HandleAtom atom);
+  FunctionNodeType fieldInitializerOpt(YieldHandling yieldHandling, Node name,
+                                       HandleAtom atom, size_t& numFieldKeys);
   FunctionNodeType synthesizeConstructor(HandleAtom className,
                                          uint32_t classNameOffset);
 